### PR TITLE
get rid of globalThis.interceptor.settings

### PR DESCRIPTION
**Pull request by KillariDev**

### CODE REVIEW

**Review by MicahZoltu — APPROVED** [review]
Huge changeset, so I'm not particularly confident in my review.  Only comments are style though, no blockers.

File: extension/app/ts/background/accessManagement.ts [hunk 1]
```diff
@@ -104,93 +103,94 @@ export function getAddressesThatDoNotNeedIndividualAccesses(settings: Settings)
...
-			if (x.addressAccess.find( (x) => x.address === address) === undefined) {
 				return {
 					website: {
 						...website,
 						icon: x.website.icon ? x.website.icon : website.icon,
 						title: x.website.title ? x.website.title : website.title,
 					},
 					access: x.access ? x.access : access,
-					addressAccess:  [ ...x.addressAccess, { address: address, access: access } ]
+					addressAccess: x.addressAccess.map( (x) => ( x.address === address ? { address: address, access: access } : x ) )
```
[the remark this answers is not in the document]
Comment (KillariDev) [reply extension/app/ts/background/accessManagement.ts:160]: no consistent style :). I am fixing these when I see them, but I am not always seeing them

File: extension/app/ts/background/popupMessageHandlers.ts [hunk 2]
```diff
@@ -73,73 +82,69 @@ export async function removeAddressBookEntry(_simulator: Simulator, removeAddres
...
 	}
 }
 
 export async function changeInterceptorAccess(_simulator: Simulator, accessChange: ChangeInterceptorAccess) {
-	if (globalThis.interceptor.settings === undefined) return
-	globalThis.interceptor.settings.websiteAccess = accessChange.options
-	saveWebsiteAccess(accessChange.options)
-	updateWebsiteApprovalAccesses()
-	sendPopupMessageToOpenWindows({ method: 'popup_interceptor_access_changed' })
+	await updateWebsiteAccess(() => accessChange.options) // TODO: update 'popup_changeInterceptorAccess' to return list of changes instead of a new list
```
[the remark this answers is not in the document]
Comment (KillariDev) [reply extension/app/ts/background/popupMessageHandlers.ts:116]: added issue about it

File: extension/app/ts/background/settings.ts [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
+	makeMeRich: 'makeMeRich',
+	addressbookTabId: 'addressbookTabId',
+	ConfirmationWindowPromise: 'ConfirmationWindowPromise',
+	ChainChangeConfirmationPromise: 'ChainChangeConfirmationPromise',
+	PersonalSignPromise: 'PersonalSignPromise',
+	InterceptorAccessRequestPromise: 'InterceptorAccessRequestPromise',
+	simulationResults: 'simulationResults',
+	signerName: 'signerName',
+	currentTabId: 'currentTabId',
+}
```
[the remark this answers is not in the document]
Comment (KillariDev) [reply extension/app/ts/background/settings.ts:30]: Good point. I am not sure why I did it like this

[the remark this answers is not in the document]
Comment (KillariDev) [reply extension/app/ts/background/settings.ts:30]: I created issue about it to avoid conflicts in the next pr's

**Pull request merged**